src/content-script/listeners/player.ts

```diff
@@ -0,0 +1,110 @@
+import { debug } from '@root/lib/utils';
+import socket from '@root/lib/socket';
+import { SocketEvents } from '@root/lib/constants/socket';
+import store from '@contentScript/store';
+
+//------------------------------------------------------------------------------
+
+// functions for each platform that can be run to get their video
+const videoLocators = {
+  youtube: () => {
+    return document.querySelector('.html5-main-video');
+  },
+  vimeo: () => {
+    return document.querySelector('video');
+  },
+  default: () => {
+    return document.querySelector('video');
+  },
+};
+
+//------------------------------------------------------------------------------
+
+function findVideoPlayer() {
+  const url = window.location.href;
+  const platforms = ['youtube', 'vimeo'];
```

Comment:
Let's strengthen types.

Can you add a new file (`lib/types/index.ts`) and create a supported platforms type? Then you can type the array accordingly because right now we're getting a type error below.

e.g.

```ts
export type SupportedPlatforms = 'youtube' | 'vimeo';
```

Reply:
Agreed on type strengthening.

I still want to iterate through them, so type will be defined with enum
```
export enum SupportedPlatforms {
  YOUTUBE = 'youtube',
  VIMEO = 'vimeo',
}
```